packages/clarity-js/src/interaction/pointer.ts

```diff
@@ -76,13 +80,19 @@ function touch(event: Event, root: Node, evt: TouchEvent): void {
                     activeTouchPointIds.delete(id);
                     break;
             }
-            const isPrimary = activeTouchPointId === id;
+            const isPrimary = hasPrimaryTouch && primaryTouchId === id;
 
             // Check for null values before processing this event
             if (x !== null && y !== null) { handler({ time: t, event, data: { target: target(evt), x, y, id, isPrimary } }); }
+
+            // Reset primary touch point id once touch event ends
+            if (event === Event.TouchCancel || event === Event.TouchEnd) {
+                if (primaryTouchId === id) { hasPrimaryTouch = false; }
```

Comment:
reset primarytouchId here

Reply:
wouldnt matter as the value we reset it to might end up matching the new touch id, hence the use of hasPrimaryTouch boolean